ThreeDSecure/src/main/java/com/braintreepayments/api/CardinalClient.java

```diff
@@ -58,6 +58,8 @@ void continueLookup(ThreeDSecureResult threeDSecureResult, CardinalChallengeObse
         } catch (RuntimeException e) {
             throw new BraintreeException("Cardinal SDK cca_continue Error.", e);
         }
+
+        Cardinal.getInstance().cleanup();
```

Comment:
I'm wondering if this should be invoked in the `onValidated` callback (line 37) instead? But if we tested and confirmed it resolves the leak when called here that's good with me too

Reply:
When I was testing I noticed there was still a memory leak with it in `onValidated` since we call `Cardinal.getInstance().init` afterwards. I think actually having it down here may resolve both memory leaks but maybe I am doing something wrong with Leak Canary.